src/robot/running/librarykeywordrunner.py

```diff
@@ -104,7 +106,13 @@ def _run_with_output_captured_and_signal_monitor(self, runner, context):
     def _run_with_signal_monitoring(self, runner, context):
         try:
             STOP_SIGNAL_MONITOR.start_running_keyword(context.in_teardown)
-            return runner()
+            runner_result = runner()
+            if asyncio.iscoroutine(runner_result):
+                try:
+                    asyncio.get_running_loop()
+                except RuntimeError:
+                    runner_result = context.event_loop.run_until_complete(runner_result)
+            return runner_result
```

Comment:
What's the point of [asyncio.get_running_loop()](https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_running_loop) here? It looks like it would be used for checking is there a running loop, but I would have though that the possibly returned loop would then be used for something.

There's also a problem that `asyncio.get_running_loop()` is new in Python 3.7 and we still support Python 3.6. I don't mind our new async support not working with Python 3.6, but there should be an explicit error in that case.

Reply:
The check is important because we can run keywords inside an async function, like the builtIn example. So if we try to make the event loop run again, it will throw an error.  So if a running loop exists, we just need to return the coroutine. I would like to have it together with the If condition, but there's no function to check without raising an exception. I can make a wrapper and name it is_loop_running or something like that if the intention is not clear.

Reply:
I see. The thing that looks strange is just returning the `runner_result` if there's no exception. It's clear there's an event loop in that case, but I would have though you need to then use that loop for something here explicitly. Perhaps a comment could be added to explain this.

As I wrote above, I'm fine making this a Python 3.7+ feature but then there should be an explicit exception. On Slack you mentioned that you found some private `asyncio` method that exists also in Python 3.6 and could be used here instead. I don't think using a private method is good long term solution, but it could certainly used if `get_running_loop()` doesn't exist. In that case checking is there a loop should certainly be done in a helper method.